src/app/(shared-nav)/courses/page.tsx

```diff
@@ -0,0 +1,101 @@
+"use client"
+import { languages } from "@/app/enums/language.enum";
+import { levels } from "@/app/enums/level.enum";
+import { CoursCard } from "@/components/coursCard";
+import { coursDetails } from "@/types/coursDetails.type";
+import { ChangeEvent, useEffect, useState } from "react";
+import { IoSearch } from "react-icons/io5";
+
+export default function Courses(){
+    const [searchText, setSearchText] = useState<String>("");
+    const [courses, setCourses] = useState<coursDetails[]>(DATA);
+    const handleSearch = (event: ChangeEvent<HTMLInputElement>)=>{
+        setSearchText(event.target.value);
+    }
+    useEffect(()=>{
+        if(searchText !== "")
+            setCourses(DATA.filter(cours=> cours.title.includes(searchText as string)))
+        else if([null, undefined, ""].includes(searchText as (string | null | undefined))){
+            //force to set the value of searchText state to be an empty string even the if the stored value is undefined or null
+            setSearchText("");
+            setCourses(DATA);
+        }
+    }, [searchText]);
+    const handleSuggestion = (title: string)=>{
+        setSearchText(title);
+    }
+    return <div className="text-black w-screen flex-1 flex justify-start items-center flex-col gap-10 pt-10">
+        {/* search input */}
+        <div className="p-2 border-[2px] border-black rounded-lg flex justify-center items-center gap-3 w-[340px]">
+            <input placeholder="برمجة"
+                   type="text"
+                   className="text-center text-black font-bold placeholder-black focus:outline-none"
+                   onChange={handleSearch}
+                   value={searchText as string}
+            />
+            <IoSearch className="w-[25px] h-[25px] font-bold"/>
+        </div>
+        {/* best keywords */}
+        <div className="flex flex-col gap-5 items-end">
+            <p>متداول الان</p>
+            <div className="flex gap-3">
+                <button onClick={()=>handleSuggestion("تعلم الآلة")} className="border border-black py-2 px-4 font-bold bg-[#FF0000] text-white rounded-md">تعلم الآلة</button>
```

Comment:
Every ()=> handleSuggestion 
will create a function from scratch which is not so good from a performance perspective. 

make the function handle suggestion a callback function. 


const handleSuggestion = useCallback((suggestion) => {
  // Your handleSuggestion logic here
}, []);

You can also turn these options into an array and use map since the only thing changing is the words. 

const suggestions = ["تعلم الآلة", "الخوارزميات", "اساسيات البرمجة"];

    {suggestions.map((suggestion) => (
      <button
        key={suggestion}
        onClick={() => handleSuggestion(suggestion)}
        className={'border border-black py-2 px-4 font-bold bg-[#FF0000] text-white rounded-md'}
      >
        {suggestion}
      </button>
    ))}

Reply:
inline arrow function, such as onClick={() => handleSuggestion(suggestion)}, will create a new function on each render.